### ISSUE TITLE
.p-* class names conflict with Microformats

### ISSUE THREAD

**Issue by dissolve** [twbs/bootstrap#20185]
Microformats uses one and two letter prefixings to indicate parsing rules for html.

http://microformats.org/wiki/microformats2
```
'h-*' for root class names, e.g. 'h-card'
'p-*' for plain (text) properties, e.g. 'p-name'
'u-*' for URL properties, e.g. 'u-photo'
'dt-*' for date/time properties, e.g. 'dt-bday'
'e-*' for embedded markup properties, e.g. 'e-note'.
```
(also x- for experimental)

Obviously the new p- prefixes clash with this and will distort parsing of any website using the new prefixes.  And these prefixes are already quite well established on 10s of thousands of sites (they have been using it since 2009)

**Comment by cvrebert** [comment]
@mdo How would you feel about changing these to e.g. `.pos-*` ?

**Comment by patrickhlauke** [comment]
this is where "let's just (ab)use class names" rather than using a custom attribute/`data-` attribute comes back to bite the microformats project...but in principle if it's no big problem for us it's probably a nice courtesy to not immediately break their stuff...

**Comment by kevinmarks** [comment]
Using utility classes like this to effectively add inline styles is initially convenient, but does lead to messy markup over time. Rather than p- and m- prefixes why not padding- and margin- ?
I wrote about this tendency of pretending classes like this aren't inline style tags:  http://www.kevinmarks.com/u-means-style.html

**Comment by calvinf** [comment]
After gzipping, the longer classnames shouldn't have a big impact on HTML page size or the minified CSS.

**Comment by mdo** [comment]
Partially addressed in #20684. No further changes planned to those classes though, so there is still some conflict.
